### PR TITLE
Implement error::Error for TimerError

### PR DESCRIPTION
**Pull request by mnussbaum**
I'm currently trying to wrap Mio for a coroutine library, and this will help me simplify my error handling. I think it is also a plus for Mio to implement the standard error traits as widely as possible.

### CODE REVIEW

File: src/timer.rs [hunk 1]
```diff
@@ -303,6 +309,16 @@ impl TimerError {
     }
 }
 
+impl error::Error for TimerError {
+    fn description(&self) -> &str {
+        self.desc
+    }
+
+    fn cause(&self) -> Option<&error::Error> {
```
Comment (reem) [inline src/timer.rs]: This method has a default impl, so you don't have to re-implement it.

File: src/timer.rs [hunk 2]
```diff
@@ -294,6 +294,12 @@ pub struct TimerError {
     desc: &'static str,
 }
 
+impl fmt::Display for TimerError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        write!(fmt, "{}", self.desc)
```
Comment (reem) [inline src/timer.rs]: It would be nice to include the `TimerErrorKind` in this output as well, not just the description.

**Comment by mnussbaum** [comment]
Thanks for the fast feedback! Removed the redundant `cause` and added the kind into the display.

**Comment by mnussbaum** [comment]
Closing as I believe this was merged in cf8a9a2e5cd5f4c5359caf147ec175c254372b15

**Pull request closed without merging**